zebra-chain/src/orchard/action.rs

```diff
@@ -0,0 +1,69 @@
+use std::io;
+
+use halo2::pasta::pallas;
+
+use crate::{
+    primitives::redpallas::{self, SpendAuth},
+    serialization::{
+        serde_helpers, ReadZcashExt, SerializationError, ZcashDeserialize, ZcashSerialize,
+    },
+};
+
+use super::{
+    commitment::{self, ValueCommitment},
+    keys,
+    note::{self, Nullifier},
+};
+
+/// An Action description, as described in the [Zcash specification §7.3][actiondesc].
+///
+/// Action transfers can optionally perform a spend, and optionally perform an
+/// output.  Action descriptions are data included in a transaction that
+/// describe Action transfers.
+///
+/// [actiondesc]: https://zips.z.cash/protocol/nu5.pdf#actiondesc
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Action {
+    /// A value commitment to net value of the input note minus the output note
+    pub cv: commitment::ValueCommitment,
+    /// The nullifier of the input note being spent.
+    pub nullifier: note::Nullifier,
+    /// The randomized validating key for spendAuthSig,
+    pub rk: redpallas::VerificationKeyBytes<SpendAuth>,
+    /// The x-coordinate of the note commitment for the output note.
+    #[serde(with = "serde_helpers::Base")]
+    pub cm_x: pallas::Base,
+    /// An encoding of an ephemeral Pallas public key.
```

Comment:
I'm assuming this is output-related?
```suggestion
    /// An encoding of an ephemeral Pallas public key used to encrypt the output note.
```